package/MDAnalysis/analysis/hole2/helper.py

```diff
@@ -0,0 +1,556 @@
+        HOLE executable
+
+
+    Returns
+    -------
+    str
+        Output file name
+    """
+    with open(outfile, 'w') as output:
+        proc = subprocess.Popen(executable, stdin=subprocess.PIPE,
+                                stdout=output)
+        stdout, stderr = proc.communicate(infile_text.encode('utf-8'))
+
+    # check output in case of errors
+    with open(outfile, 'r') as output:
+        for line in output:
+            if line.strip().startswith(('*** ERROR ***', 'ERROR')):
+                proc.returncode = 255
+                break
+
+    # die in case of error
+    if proc.returncode != 0:
+        msg = 'HOLE failure ({}). Check output {}'
+        logger.fatal(msg.format(proc.returncode, outfile))
+        if stderr is not None:
+            logger.fatal(stderr)
+        raise ApplicationError(proc.returncode,
+                               msg.format(executable, outfile))
+
+    logger.info('HOLE finished. Output: {}'.format(outfile))
+    return outfile
+
+
+def collect_hole(outfile='hole.out'):
+    """Collect data from HOLE output
+
+    Parameters
+    ----------
+    outfile: str, optional
+        HOLE output file to read. Default: 'hole.out'
+
+
+    Returns
+    -------
+    dict
+        Dictionary of HOLE profiles as record arrays
+    """
+    fmt = util.FORTRANReader('3F12')
+    recarrays = {}
+
+    with open(outfile, 'r') as output:
+        toggle_read = False
+        profile = 0
+        records = []
+        for line in output:
+            line = line.rstrip()  # preserve columns in FORTRAN output
+            # multiple frames from dcd in?
+            if line.startswith(" Starting calculation for position number"):
+                fields = line.split()
+                profile = int(fields[5])
+                records = []
+                logger.debug('Started reading profile {}'.format(profile))
+                continue
+
+            # found data
+            if line.startswith(' cenxyz.cvec'):
+                toggle_read = True
+                logger.debug('Started reading data')
+                continue
+
+            if toggle_read:
+                if len(line.strip()) != 0:
+                    try:
+                        rxncoord, radius, cenlineD = fmt.read(line)
+                    except:
+                        msg = 'Problem parsing line: {}. Check output file {}'
+                        logger.exception(msg.format(line, outfile))
+                        raise
+                    records.append((rxncoord, radius, cenlineD))
+                    continue
+                # end of data
+                else:
+                    toggle_read = False
+                    names = ['rxn_coord', 'radius', 'cen_line_D']
+                    recarr = np.rec.fromrecords(records,
+                                                formats='f8, f8, f8',
+                                                names=names)
+                    recarrays[profile] = recarr
```

Comment:
Is it worth breaking the loop here? To avoid iterating further lines in the file. Or can there be multiple sections to read?

Reply:
This is for reading CHARMM DCD files using the hole function, although there is not current test for this that works on Travis (the actual HOLE program breaks on them, despite my local version working)